test/parallel/test-message-channel-sharedarraybuffer.js

```diff
@@ -0,0 +1,28 @@
+/*global SharedArrayBuffer*/
+'use strict';
+// Flags: --harmony-sharedarraybuffer --expose-gc
+
+const common = require('../common');
+const assert = require('assert');
+const { Worker } = require('worker');
+
+{
+  const sharedArrayBuffer = new SharedArrayBuffer(12);
+  const local = Buffer.from(sharedArrayBuffer);
+
+  const w = new Worker(`
+    require('worker').on('workerMessage', ({ sharedArrayBuffer }) => {
+      const local = Buffer.from(sharedArrayBuffer);
+      local.write('world!', 6);
+      require('worker').postMessage();
+    });
+  `, { eval: true });
+  w.on('message', common.mustCall(() => {
+    assert.strictEqual(local.toString(), 'Hello world!');
+    global.gc();
+    w.terminate();
+  }));
+  w.postMessage({ sharedArrayBuffer });
```

Comment:
Anything wrong with just posting the Buffer directly?

Reply:
Not a technical one … one some level, just wanted to make sure it works even when the SAB occurs *somewhere* within the passed object.

The real reason is that this kind of test stops me from thinking it would be possible to get a list of all SABs inside the passed object before serialization and making changes to the message passing code based on that :P